packages/presentation/presentation_publishing/interfaces/has_edit_capabilities.ts

```diff
@@ -15,9 +15,9 @@ import { HasTypeDisplayName } from './has_type';
  * edited, and an isEditingEnabled function.
  */
 export interface HasEditCapabilities extends HasTypeDisplayName {
-  onEdit: () => void;
+  onEdit: () => Promise<void>;
   isEditingEnabled: () => boolean;
-  getEditHref?: () => string | undefined;
```

Comment:
Both `onEdit` and `getEditHref` should be `async`:
- `onEdit` often requires opening a flyout and waiting for input from the user - in the image embeddable, for example, the implementation of `onEdit` was async but (for some reason) was not throwing a linting error.
- `getEditHref` may use the locator `getUrl` method, which is `async` - also, the `getHref` method in the `EditPanelAction` is already `async`, so we just need to await the call for `getEditHref`